### ISSUE TITLE
Error thrown for single object

### ISSUE THREAD

**Issue by dougrichardson** [dougiesquire/xbootstrap#7]
When a single object is passed to `block_bootstrap`, the following error is sometimes thrown:

```
import numpy as np
import xarray as xr
from xbootstrap import block_bootstrap

samples_array = np.random.normal(size=(150, 200, 65))

samples = xr.DataArray(
    samples_array,
    dims=["lat", "lon", "time"],
    coords={"lat": range(150), "lon": range(200), "time": range(65)}
)

samples = samples.chunk()

bootstraps = block_bootstrap(
    samples,
    blocks={'time': 1},
    n_iteration=100,
    exclude_dims=None,
    circular=True
)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
File <timed exec>:1

Cell In[119], line 405, in block_bootstrap(blocks, n_iteration, exclude_dims, circular, *objects)
    404 if len(objects) == 1:
--> 405     return xr.concat(
    406         *bootstraps,
    407         dim="iteration",
    408         coords="minimal",
    409         compat="override",
    410     )
    411 else:
    412     return tuple(
    413         [
    414             xr.concat(
   (...)
    421         ]
    422     )

TypeError: concat() got multiple values for argument 'dim'
```

This appears to be related to how the function splits up the desired number of iterations. In this example, some arrays have an iteration dimension size of 17, others have 14. But I can't figure out exactly what's going wrong - it's been fine in other uses for me.

A simple adjustment of where the `if len(objects) == 1` statements lies in the function would solve this and I can submit a PR if desired.

**Comment by dougiesquire** [comment]
Thanks for reporting @dougrichardson. The issue comes about when there is a single dask-backed object that is chunked in a way that means it is processed in pieces within xbootstrap. I agree that refactoring to move the `if` statement would fix the issue.

If you'd like to submit a PR, it would be great if you could add a test to `xbootstrap/tests` that fails with the current code, but passes once you've added the change. Let me know if you haven't got time for this and I can do it :)

**Comment by dougiesquire** [comment]
Closed by #8
